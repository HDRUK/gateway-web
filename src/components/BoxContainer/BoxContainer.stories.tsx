import type { Meta, StoryObj } from "@storybook/react";
import Box from "@/components/Box";
import BoxContainer from "./BoxContainer";

/**
 * Mui documentation:
 * https://mui.com/material-ui/react-box/
 */

const meta: Meta<typeof BoxContainer> = {
    component: BoxContainer,
    title: "Layout/BoxContainer",
    tags: ["autodocs"],
};

export default meta;

type Story = StoryObj<typeof BoxContainer>;

export const Columns: Story = {
    render: () => (
        <BoxContainer sx={{ gridTemplateColumns: "repeat(3, 1fr)" }}>
            <Box>Container 1</Box>
            <Box>Container 2</Box>
            <Box>Container 3</Box>
        </BoxContainer>
    ),
};
export const Responsive: Story = {
    render: () => (
        <BoxContainer
            sx={{
                gridTemplateColumns: {
                    mobile: "repeat(1, 1fr)",
                    tablet: "repeat(5, 1fr)",
                },
                gap: {
                    mobile: 0,
                    tablet: 1,
                },
            }}>
            <Box sx={{ gridColumn: { tablet: "span 2", laptop: "span 1" } }}>
                Container 1
            </Box>
            <Box sx={{ gridColumn: { tablet: "span 3", laptop: "span 4" } }}>
                Container 2
            </Box>
        </BoxContainer>
    ),
};
