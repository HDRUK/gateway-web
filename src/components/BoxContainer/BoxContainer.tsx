import MuiBox, { BoxProps as MuiBoxProps } from "@mui/material/Box";

/**
 * Documents:
 * https://mui.com/material-ui/react-box/
 */

const BoxContainer = (props: MuiBoxProps) => {
    const { sx, ...rest } = props;
    return (
        <MuiBox
            sx={{
                gap: 1,
                display: "grid",
                ...sx,
            }}
            {...rest}
        />
    );
};

export default BoxContainer;
