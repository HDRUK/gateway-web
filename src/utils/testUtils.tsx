import React, { ReactNode } from "react";
import {
    render,
    renderHook,
    RenderHookOptions,
    RenderOptions,
    RenderResult,
    RenderHookResult,
} from "@testing-library/react";
import { SWRConfig } from "swr";
import { CacheProvider, ThemeProvider } from "@emotion/react";
import ActionBarProvider from "@/providers/ActionBarProvider";
import DialogProvider from "@/providers/DialogProvider";
import theme from "@/config/theme";
import createEmotionCache from "@/config/createEmotionCache";

const clientSideEmotionCache = createEmotionCache();

const Wrapper = ({ children }: { children: ReactNode }) => {
    return (
        <SWRConfig
            value={{
                provider: () => new Map(),
            }}>
            <CacheProvider value={clientSideEmotionCache}>
                <ThemeProvider theme={theme}>
                    <ActionBarProvider>
                        <DialogProvider>{children}</DialogProvider>
                    </ActionBarProvider>
                </ThemeProvider>
            </CacheProvider>
        </SWRConfig>
    );
};

interface OptionProps
    extends RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRender = (
    ui: React.ReactElement<
        unknown,
        string | React.JSXElementConstructor<unknown>
    >,
    options?: OptionProps
): RenderResult => {
    const { wrapperProps, ...rest } = options || {};

    return render(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

interface OptionHookProps
    extends RenderHookOptions<
        unknown,
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRenderHook = (
    ui: (initialProps: unknown) => unknown,
    options?: OptionHookProps
): RenderHookResult<unknown, unknown> => {
    const { wrapperProps, ...rest } = options || {};
    return renderHook(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

export * from "@testing-library/react";

export { customRenderHook as renderHook };
export { customRender as render };
