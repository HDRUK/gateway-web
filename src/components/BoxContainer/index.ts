import BoxContainer from "./BoxContainer";

export default BoxContainer;
