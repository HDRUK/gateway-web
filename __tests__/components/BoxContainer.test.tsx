import React from "react";
import BoxContainer from "@/components/BoxContainer";
import { render } from "../testUtils";

describe("BoxContainer", () => {
    it("should render component", async () => {
        const wrapper = render(
            <BoxContainer>BoxContainer content</BoxContainer>
        );

        expect(wrapper.container).toMatchSnapshot();
    });
});
