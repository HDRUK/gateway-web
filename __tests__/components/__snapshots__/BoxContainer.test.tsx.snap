// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxContainer should render component 1`] = `
<div>
  <div
    class="MuiBox-root css-5fbagg"
  >
    BoxContainer content
  </div>
</div>
`;
